### PR TITLE
Feat mod racon

### PR DESCRIPTION
**Pull request by MunissaSadykova**

### CODE REVIEW

File: configs/module-defaults.config [hunk 1]
```diff
@@ -110,7 +110,7 @@ params {
             suffix           = ""
             publish_dir      = "minionqc"
         }
-	'nanoplot' {
+				'nanoplot' {
```
Comment (chris-cheshire) [inline configs/module-defaults.config:113]: indent needs fixing

File: tools/racon/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM nfcore/base:1.7
+LABEL authors="munissa.sadykova@oist.jp" \
+      description="Docker image containing all requirements for the nf-modules-racon module"
+
+# Install apt packages
+RUN apt-get update \
+	&& apt-get clean \
+	&& rm -rf /var/lib/apt/lists/*
```
Comment (chris-cheshire) [inline tools/racon/Dockerfile:8]: you dont need this, its not installing anything

File: tools/racon/main.nf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// Process definition
+process racon {
+    publishDir "${params.outdir}/${opts.publish_dir}",
+        mode: "copy",
+        overwrite: true,
+        saveAs: { filename ->
+                      if (opts.publish_results == "none") null
+                      else filename }
+
+    container "luslab/nf-modules-racon:latest"
```
Comment (chris-cheshire) [inline tools/racon/main.nf:15]: are you sure there isnt a biocontainer for this?

File: tools/racon/main.nf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    input:
+        val opts
+        tuple val(meta), path(reads), path(overlap_paf), path(assembly_fasta)
```
Comment (chris-cheshire) [inline tools/racon/main.nf:19]: this is generally not the correct structure - you want to pass these files as separate channels unless they are directly linked to the reads file?

File: tools/racon/test/nextflow.config [hunk 5]
```diff
@@ -0,0 +1,5 @@
+includeConfig "$baseDir/../../../configs/module-defaults.config"
+
+params.outdir = "./results"
+docker.enabled = true
+
```
Comment (chris-cheshire) [inline tools/racon/test/nextflow.config:5]: remove line

File: tools/racon/test/main.nf [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+/* Run tests
+--------------------------------------------------------------------------------------*/
+
+workflow {
+    // Run racon on the test set
+    racon(params.modules['racon'], ch_racon_data)
+
+    // Collect file names and view output
+    racon.out.fasta | view
+
```
Comment (chris-cheshire) [inline tools/racon/test/main.nf:48]: remove line

File: tools/racon/test/main.nf [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+/*------------------------------------------------------------------------------------*/
+/* Run tests
+--------------------------------------------------------------------------------------*/
+
+workflow {
+    // Run racon on the test set
+    racon(params.modules['racon'], ch_racon_data)
+
+    // Collect file names and view output
+    racon.out.fasta | view
```
Comment (chris-cheshire) [inline tools/racon/test/main.nf:47]: no channel count assertion

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

**Comment by mjmansfi** [comment]
The `minionqc` module is failing because it points to a folder with a typo that has been fixed in this branch. Please update the minionQC `main.nf`.

File: tools/racon/main.nf [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+                      if (opts.publish_results == "none") null
+                      else filename }
+
+    container "quay.io/biocontainers/racon:1.4.12--he513fc3_0"
+
+    input:
+        val opts
+        tuple val(meta), path(reads)
+				path(overlap_paf)
+				path(assembly_fasta)
```
Comment (chris-cheshire) [inline tools/racon/main.nf:21]: Tidy up indentation but good otherwise!

**Comment by chris-cheshire** [comment]
You still need to add this to the test schedule

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

**Review by chris-cheshire — APPROVED** [review]

**Pull request merged**